File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.py
import itertools

def solution(numbers):
    answer = []
    
    num_list = list(itertools.combinations(numbers, 2))
    
    for num in num_list:
        if sum(num) not in answer:
            answer.append(sum(num))
            
    answer.sort()
    
    return answer

File: 프로그래머스/1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.js
function solution(numbers) {
    let answer = new Set();
    
    const dfs = (currentArray, startIndex) => {
        if (currentArray.length === 2) {
            answer.add(currentArray.reduce((sum, num) => sum + num, 0));
            return;
        }
        
        for (let i = startIndex; i < numbers.length; i++) {
            dfs([...currentArray, numbers[i]], i + 1);
        }
    }
    
    dfs([], 0);
    
    return [...answer].sort((a, b) => a - b);
}